services/rfq/relayer/inventory/rebalance.go

```diff
@@ -27,183 +27,185 @@ type RebalanceManager interface {
 	Execute(ctx context.Context, rebalance *RebalanceData) error
 }
 
-// getRebalance builds a rebalance action based on current token balances and configured thresholds.
-// Note that only the given chain/token pair is considered for rebalance (as the destination chain).
-//
-//nolint:cyclop,nilnil
-func getRebalance(span trace.Span, cfg relconfig.Config, tokens map[int]map[common.Address]*TokenMetadata, chainID int, token common.Address) (rebalance *RebalanceData, err error) {
-	// get rebalance method
-	methods, err := cfg.GetRebalanceMethods(chainID, token.Hex())
+// getRebalances gets the best rebalance action for each token.
+func getRebalances(ctx context.Context, cfg relconfig.Config, inv map[int]map[common.Address]*TokenMetadata) (rebalances map[string]*RebalanceData, err error) {
+	rebalances = map[string]*RebalanceData{}
+
+	rebalanceCandidates, err := getRebalanceCandidates(ctx, cfg, inv)
 	if err != nil {
-		return nil, fmt.Errorf("could not get rebalance method: %w", err)
-	}
-	if len(methods) == 0 {
-		return nil, nil
+		return nil, fmt.Errorf("could not get rebalance candidates: %w", err)
 	}
 
-	// get token metadata
-	var rebalanceTokenData *TokenMetadata
-	for address, tokenData := range tokens[chainID] {
-		if address == token {
-			rebalanceTokenData = tokenData
-			break
+	for tokenName, methodCandidates := range rebalanceCandidates {
+		methodCandidatesSlice := []RebalanceData{}
+		for _, candidate := range methodCandidates {
+			if candidate == nil {
+				continue
+			}
+			methodCandidatesSlice = append(methodCandidatesSlice, *candidate)
 		}
+		rebalances[tokenName] = getBestRebalance(methodCandidatesSlice)
 	}
 
-	// evaluate the origin and dest of the rebalance based on min/max token balances
-	originTokenData, destTokenData, method := getRebalanceMetadatas(cfg, tokens, rebalanceTokenData.Name, methods)
-	if originTokenData == nil {
-		if span != nil {
-			span.SetAttributes(attribute.Bool("no_rebalance_origin", true))
-		}
-		return nil, nil
-	}
-	if destTokenData == nil {
-		if span != nil {
-			span.SetAttributes(attribute.Bool("no_rebalance_dest", true))
-		}
-		return nil, nil
-	}
-	if method == relconfig.RebalanceMethodNone {
-		if span != nil {
-			span.SetAttributes(attribute.Bool("no_rebalance_method", true))
-		}
-		return nil, nil
-	}
+	return rebalances, nil
+}
 
-	// if the given chain is not the destination of the rebalance, no need to do anything
-	if destTokenData.ChainID != chainID {
-		if span != nil {
-			span.SetAttributes(attribute.Int("rebalance_dest", destTokenData.ChainID))
+// getRebalanceCandidates gets the best rebalance for each token and rebalance method supported by the config.
```

Comment:
I believe the documentation should state that it gets _all possible rebalances_ for each token rather than _the best rebalance_

Reply:
Perhaps a misleading function name- the intention is to get the best rebalance (taking origin / dest balances into account) for each token,method pair, to be filtered down across methods in `getRebalances()`. Note how `getRebalanceCandidates()` returns map[int]map[method]Rebalance, instead of the final value being a collection of rebalances

Reply:
Oh, I see what you mean now, thanks. The name seems fine to me, just needed to reread the docs/impl one more time to get this.